docs/sql-pyspark-comparison-with-pandas.md

```diff
@@ -0,0 +1,401 @@
+    it to another DataFrame whereas Pandas DataFrame is mutable which directly updates the state
+    of it. Typical example is `String` vs `StringBuilder` in Java.
+  
+4. PySpark operations on DataFrame tend to comply SQL.
+
+    It causes some subtleties comparing to Pandas, for instance, about `NaN`, `None` and `NULL`.
+
+There are similarities and differences between them which might bring confusion. In this document
+these are described and illuminated by several examples.
+
+
+
+## DataFrame APIs
+
+This chapter describes DataFrame APIs in both PySpark and Pandas.
+
+
+### Quick References
+
+| PySpark                                                            | Pandas                                   |
+| ------------------------------------------------------------------ | ---------------------------------------- |
+| `df.limit(3)`                                                      | `df.head(3)`                             |
+| `df.filter("a == 1 AND b == 2")`                                   | `df.filter("(df.a == 1) & (df.b == 2)")` |
+| `df.filter((df.a == 1) & (df.b == 2))`                             | `df[(df.a == 1) & (df.b == 2)]`          |
+| `df.select("a", "b")`                                              | `df[["a", "b"]]`                         |
+| `df.drop_duplicates()`                                             | `df.drop_duplicates()`                   |
+| `df.sample(fraction=0.01)`                                         | `df.sample(frac=0.01)`                   |
+| `df.groupby("a").count()`                                          | `df.groupby("a").size()`                 |
+| `df.groupby("a").agg({"b": "sum"})`                                | `df.groupby("a").agg({"b": np.sum})`     |
+| `df1.join(df2, on="a")`                                            | `pandas.merge(df1, df2, on="a")`         |
+| `df1.union(df2)`                                                   | `pandas.concat(df1, df2)`                |
+| `df = df.select(when(df["a"] < 5, df["a"] * 2).otherwise(df["a"]))`| `df.loc[pdf['a'] < 5, 'a'] *= 2`         |
+
+
+### Create DataFrame
+
+In order to create DataFrame in PySpark and Pandas, you can run the codes below: 
+
+```python
+# PySpark
+data = zip(['Chicago', 'San Francisco', 'New York City'], range(1, 4))
+spark.createDataFrame(list(data), ["city", "rank"])
+```
+
+```python
+# Pandas
+data = {'city': ['Chicago', 'San Francisco', 'New York City'], 'rank': range(1, 4)}
+pandas.DataFrame(data)
+```
+
+One notable difference when creating DataFrame is that Pandas accepts the data as below:
+
+```
+data = {
+    'city': ['Chicago', 'San Francisco', 'New York City'],
+    'rank': range(1, 4)
+}
+```
+
+and it interprets as:
+
+```
+            city  rank
+0        Chicago     1
+1  San Francisco     2
+2  New York City     3
+```
+
+So, a dictionary that contains key and multiple values becomes DataFrame but PySpark does
```

Comment:
key and ... -> multiple values for a key
becomes DataFrame -> becomes a DataFrame,
Pyspark does -> Pyspark does not?